samples/fullduplex/Core_8/Shared/StoreAuditBodyInAzureBlobStorageBehavior.cs

```diff
@@ -0,0 +1,57 @@
+﻿namespace Shared
+{
+    using Azure.Storage.Blobs;
+    using Azure.Storage.Blobs.Models;
+    using Microsoft.Toolkit.HighPerformance;
+    using NServiceBus;
+    using NServiceBus.Pipeline;
+    using NServiceBus.Routing;
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    public class StoreAuditBodyInAzureBlobStorageBehavior : Behavior<IDispatchContext>
+    {
+        public StoreAuditBodyInAzureBlobStorageBehavior(BlobContainerClient blobContainerClient)
+        {
+            this.blobContainerClient = blobContainerClient;
+        }
+
+        public override async Task Invoke(IDispatchContext context, Func<Task> next)
+        {
+            foreach (var operation in context.Operations)
+            {
+                var unicastAddress = operation.AddressTag as UnicastAddressTag;
+
+                if (unicastAddress?.Destination != "audit")
+                {
+                    continue;
+                }
+
+                var message = operation.Message;
+
+                var blob = blobContainerClient.GetBlobClient(message.MessageId);
+                var options = new BlobUploadOptions
+                {
+                    Metadata = new Dictionary<string, string>()
+                        {
+                        { "ContentType", message.Headers[Headers.ContentType] },
+                        { "BodySize", message.Body.Length.ToString()}
+                        }
+                };
+
+                await blob.UploadAsync(
+                    BinaryData.FromStream(message.Body.AsStream()),
+                    options,
+                    context.CancellationToken
+                    ).ConfigureAwait(false);
+
+                operation.Message.UpdateBody(null); //TODO: Would ReadOnlyMemory<byte>.Empty be better?
```

Comment:
We may want to consider adding a couple of header values here to indicate the body was stored prior to sending, ServiceControl uses an awkward [`BodyNotStored`](https://github.com/Particular/ServiceControl/blob/3d81d2f1d0f90db387704a856c69ae1e926b03a1/src/ServiceControl.Audit/Auditing/BodyStorage/BodyStorageFeature.cs#L56) flag to know if the body is stored with the headers or in body storage.  Also, maybe consider setting the [`ContentLength`](https://github.com/Particular/ServiceControl/blob/3d81d2f1d0f90db387704a856c69ae1e926b03a1/src/ServiceControl.Audit/Auditing/BodyStorage/BodyStorageFeature.cs#L43) as well since everything will be 0 on the other side.   As it stands, the logic on the ServiceControl end is a bit fuzzy for if something is or isn't stored in body storage. It is generally just assuming if the body is 0 length something else stored it.

Reply:
Yea, good idea, I'll add something. 

>ServiceControl uses an awkward BodyNotStored flag to know if the body is stored

The `BodyNotStored` feels like an SC internal thing so let go with something like `NServiceBus.BodyStorage` = `AzureBlob`

What more would we add? (Storage account name + container name?)

>Also, maybe consider setting the ContentLength as well since everything will be 0 on the other side

Will do